### PR TITLE
ffi: Simplify regular (non-OTLP) tracing initialization

### PR DESCRIPTION
**Pull request by jplatte**
Wanted to switch from tracing-android to paranoid-android, ran out of time for today.

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/platform.rs [hunk 1]
```diff
@@ -108,6 +111,19 @@ fn setup_tracing_helper(configuration: String, colors: bool) {
         .init();
 }
 
+#[cfg(target_os = "android")]
+pub fn setup_tracing(configuration: String) {
```
Comment (poljar) [inline bindings/matrix-sdk-ffi/src/platform.rs:115]: Is this one still used? Doesn't seem so?

**Review by poljar — APPROVED** [review]
Might be wrong but there seems to be a dead method now. Looks fine otherwise, nice trick with the `cfg!()` call.

File: bindings/matrix-sdk-ffi/src/platform.rs (hunk 1 above)
Comment (jplatte) [reply bindings/matrix-sdk-ffi/src/platform.rs:115]: Oh, this must be `setup_tracing_impl`. Looks like CI has an Android-shaped hole.

**Comment by jplatte** [comment]
Now the exact thing I pushed (and no some combination with other changes 🤦🏼) was tested locally using `cargo ndk check` in the ffi crates' directory (and I verified that before the latest force-push it *didn't* build that way).

**Pull request merged**